minio_async/progress.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+# MinIO Python Library for Amazon S3 Compatible Cloud Storage,
+# (C) 2018 MinIO, Inc.
+# (C) 2022 L-ING <hlf01@icloud.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module implements a progress printer while communicating with MinIO server
+:copyright:
+(C) 2018 by MinIO, Inc.
+(C) 2022 L-ING <hlf01@icloud.com>
+:license: Apache 2.0, see LICENSE for more details.
+"""
+
+
+import time
+
+_BAR_SIZE = 20
+_KILOBYTE = 1024
+_FINISHED_BAR = '#'
+_REMAINING_BAR = '-'
+
+_UNKNOWN_SIZE = '?'
+_STR_MEGABYTE = ' MB'
+
+_HOURS_OF_ELAPSED = '%d:%02d:%02d'
+_MINUTES_OF_ELAPSED = '%02d:%02d'
+
+_RATE_FORMAT = '%5.2f'
+_PERCENTAGE_FORMAT = '%3d%%'
+_HUMANINZED_FORMAT = '%0.2f'
+
+_DISPLAY_FORMAT = '|%s| %s/%s %s [elapsed: %s left: %s, %s MB/sec]'
+
+_REFRESH_CHAR = '\r'
+
+
+class Progress():
```

Comment:
Progress class mimics original Progress class from Minio. However it outputs data to stdout, which shouldn't be the case for library, I'll take this off for now, but later something like this can be added

Reply:
If you don't add it to library, don't forget to delete it's examples and docs.